### DISCUSSION TITLE
📺 Join Live Today: Ask the experts: Meet our Engineering Team and Ask Your Pressing Questions!

### DISCUSSION THREAD

**Discussion by kamalshree** [microsoft/hack-together-teams#59] [Announcements]
**Join us live today (Monday, Jun 12) at 4PM GMT/9AM PT in which you will the meet the engineering team and get all your pressing questions answered**

![#5 - June 12 - Ask the experts: Meet our Engineering Team and Ask Your Pressing Questions! ](https://github.com/microsoft/hack-together-teams/assets/3199282/d21a0738-0d95-4c65-8329-19dea140ab66)

we’re bringing together experts from the different product groups at Microsoft to answer all your questions about building apps for Microsoft Teams. We have an awesome line-up so be sure to join us. This is a unique opportunity for you to meet the people who build and manage the Microsoft Teams- and Microsoft 365 platform and tooling.

Meet our Enggineering Team, @lucabandMSFT, @davrous, Ankit Govil,@Stevenic, @sebastienlevert

Join us live and bring any and all questions you might have.

👋 See you there! https://aka.ms/hack-together-teams/session5

**Comment by aycabas** [comment]
This is going to be a fun session! 🥳

**Comment by waldekmastykarz** [comment]
Looking forward to seeing you all there! 👋

**Comment by JonoSuave** [comment]
Thanks Steven Ickman for answering my question about integrating Teams AI with my tenant's Microsoft Graph data. I'm a bit new to developing Teams bots in general. Do you have a sample where I can see the OAuth settings you mentioned to pass into the Application object? Then you also mentioned getting the token from the state.temp.value.authtoken?

**Reply by JonoSuave** [reply]
To offer further context, here's my attempt so far to sign in the user on line 71: https://github.com/JonoSuave/TutorialBot/blob/master/index.ts

I'm not really sure how to grab the turn context and state here.

**Reply by kavins14** [reply]
Hey, you don't have to call `app.authentication.signInUser(context: TurnContext, state: ApplicationTurnState );` explicitly. Here's how you can set up authentication:

```javascript
const app = new Application<ApplicationTurnState>({
    storage,
    ai: {
        planner,
	moderator,
	promptManager,
	prompt: "chat",
	history: {
	    assistantHistoryType: "text",
	},
    },
    authentication: {
        connectionName: "<CONNECTION_NAME>",
        title: "<TITLE>",
    }
});

```

![image](https://github.com/microsoft/hack-together-teams/assets/18351436/f6f867a1-2fad-452e-8b5a-d8f0a938126a)

The `authentication` property takes in an object of type `OAuthPromptSettings` from the `botbuilder-dialogs` library. You can look into that and see what all properties are required. Once you have that configured, send a message to the bot and the sign in flow should be invoked.

**Reply by JonoSuave** [reply]
Thank you that was super helpful. This is the documentation I used to add authentication to the bot: https://learn.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=userassigned%2Caadv2%2Cjavascript 
It's still a work in progress (getting following error: "`[onTurnError] unhandled error: DialogContextError: Could not find Connection Setting with name GraphConnectionLocalBot`"

I looked through the Graph documentation to try and instantiate a graph client by passing in an auth token. Is this what you had in mind:
```
app.ai.prompts.addFunction("graph", async (context, state) => {
	const authToken = state.temp.value.authToken;
	const graphEndpoint = "https://graph.microsoft.com/v1.0/me";
	return callMSGraph(graphEndpoint, authToken);
});

export function callMSGraph(endpoint, access_token) {
	const headers = new Headers();
	const bearer = "Bearer " + access_token;
	headers.append("Authorization", bearer);
	const options = {
		method: "GET",
		headers: headers,
	};
	const graphEndpoint = endpoint;

	fetch(graphEndpoint, options).then(function (response) {
		//do something with response
	});
}
```

Also in my skprompt.txt file do I just add a variable like so to let the AI know about the graph function added: `{{$graph}}`

**Comment by JonoSuave** [comment]
@kavins14 could you point me to documentation on how to correctly configure the Azure bot with the app registration with graph permissions? Am I able to test this locally? Do I have to provision resources first and use the Configurations settings within the Azure Bot provisioned? If so, do I need to update my env.local.user Bot Id and Bot Password?
